sdk/search/azure-search-documents-serializer/src/main/java/com.azure.search.documents.serializer/SearchType.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.serializer;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+public abstract class SearchType<T> implements Comparable<SearchType<T>>{
+    private final Type _type;
+
+    public SearchType() {
```

Comment:
Why did we add this class? If required we can change the interface in Core from taking a `Class<T>` to Type which would support both `Class<T>` and `ParameterizedType` which this seems to be attempting.

Reply:
By having SearchType<T>, we can hide the ugly type reflection operation behind for user.
```
Map<String, Object> mapInstance = new HashMap<>;
Type type = ((ParameterizedType)mapInstance.getClass().getGenericSuperclass()).getActualTypeArguments()[0];
```